### PR TITLE
[RHACS] Added RHACS Cloud Service architecture diagram

### PR DESCRIPTION
**Pull request by gaurav-nelson**
For https://issues.redhat.com/browse/ROX-12494

Added architecture diagram for RHACS CLoud Service.

Cherry-pick into `rhacs-docs-4.2`

### CODE REVIEW

File: modules/acscs-data-access-permissions.adoc [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+[id="acscs-data-access-permissions_{context}"]
+= Data access and permissions
+
+
+[role="_abstract"]
+Red Hat does not have access to any clusters on which you install the secured cluster services. Additionally, {product-title-managed-short} does not require permission to access the secured clusters. For example, you do not need to create new IAM policies, access roles, or API tokens.
+
+However, {product-title-managed-short} stores the data that secured cluster services send. All data is encrypted within the {product-title-managed-short}. The encryption of data within the {product-title-managed-short} platform helps to ensure the confidentiality and integrity of the data.
+
+When you install secured cluster services on a cluster, it generates data and transmits it to the {product-title-managed-short}. This data is kept secure within the {product-title-managed-short} platform, and only authorized SRE team members and systems can access this data. {product-title-managed-short} uses this data to help monitor the security and compliance of your cluster and applications and provide valuable insights and analytics that can help you optimize your deployments.
```
Comment (agantony) [inline modules/acscs-data-access-permissions.adoc:14]:
```suggestion
When you install secured cluster services on a cluster, it generates data and transmits it to the {product-title-managed-short}. This data is kept secure within the {product-title-managed-short} platform, and only authorized SRE team members and systems can access this data. {product-title-managed-short} uses this data to monitor the security and compliance of your cluster and applications, and to provide valuable insights and analytics that can help you optimize your deployments.
```

File: modules/acscs-secured-cluster-services-overview.adoc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Module included in the following assemblies:
+//
+// * architecture/acs-architecture.adoc
+:_content-type: CONCEPT
+[id="acscs-secured-cluster-services_{context}"]
+= Secured cluster services
+
+You install the secured cluster services on every cluster you want to secure by using the {product-title-managed-short}.
```
Comment (agantony) [inline modules/acscs-secured-cluster-services-overview.adoc:8]:
```suggestion
You install the secured cluster services on each cluster that you want to secure by using the {product-title-managed-short}.
```

File: modules/acscs-secured-cluster-services-overview.adoc [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+* *Sensor*: Sensor is the service responsible for analyzing and monitoring the cluster.
+Sensor listens to the {ocp} or Kubernetes API and Collector events to report the current state of the cluster.
+Sensor also triggers deploy-time and runtime violations based on {product-title-managed-short} policies. Additionally, Sensor is responsible for all cluster interactions, such as applying network policies, initiating reprocessing of {product-title-managed-short} policies, and interacting with the Admission controller.
+* *Admission controller*: The Admission controller prevents users from creating workloads that violate security policies in {product-title-managed-short}.
+* *Collector*: Collector analyzes and monitors container activity on cluster nodes.
+It collects container runtime and network activity information and sends the collected data to Sensor.
+* *Scanner*: In Kubernetes, the secured cluster services include Scanner-slim as an optional component.
+On {ocp}, however, {product-title-managed-short} installs a Scanner-slim version on each secured cluster to scan images in the {ocp} integrated registry and optionally other registries.
```
Comment (agantony) [inline modules/acscs-secured-cluster-services-overview.adoc:18]:
```suggestion
* *Sensor*: Sensor is the service responsible for analyzing and monitoring the cluster.
Sensor listens to the {ocp} or Kubernetes API and Collector events to report the current state of the cluster.
Sensor also triggers deploy-time and runtime violations based on {product-title-managed-short} policies. In addition, Sensor is responsible for all cluster interactions, such as applying network policies, initiating reprocessing of {product-title-managed-short} policies, and interacting with the Admission controller.
* *Admission controller*: The Admission controller prevents users from creating workloads that violate security policies in {product-title-managed-short}.
* *Collector*: Collector analyzes and monitors container activity on cluster nodes.
It collects container runtime and network activity information and sends the collected data to Sensor.
* *Scanner*: In Kubernetes, the secured cluster services include Scanner-slim as an optional component.
However, on {ocp}, {product-title-managed-short} installs a Scanner-slim version on each secured cluster to scan images in the {ocp} integrated registry and optionally other registries.
```

**Review by agantony — COMMENTED** [review]
Added suggestions; overall lgtm!

File: modules/acscs-architecture-overview.adoc [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// Module included in the following assemblies:
+//
+// * cloud_service/acscs-architecture.adoc
```
[the remark this answers is not in the document]
Comment (gaurav-nelson) [reply modules/acscs-architecture-overview.adoc:3]: Good catch, I originally placed it under cloud service topics but then moved it to architecture :)

**Pull request merged**